### PR TITLE
Multiple singlenorms

### PR DESCRIPTION
**Pull request by leonie-koch**
Changelog:

- Update the UI according to designs
- Wire up the grouped norm components with the backend logic and the data flow
- Abbreviation update does not trigger emit -> effect is, that the cancel and delete button not visible in initial 'creation' mode
- Updating the norms summary logic
- Fixing the display of norms summary, with no/ empty single norms
- Adding same abbreviation twice not possible

Design: https://www.figma.com/file/oyeWgl6BRJA15tkGjUD6Ze/neuRIS-Rechtsprechung?type=design&node-id=12695-17801&mode=design&t=ahymZby7k8JEMl1n-0

Ticket: https://digitalservicebund.atlassian.net/jira/software/c/projects/RISDEV/boards/7?assignee=62bad547ed036549273cd7de&selectedIssue=RISDEV-3546

### CODE REVIEW

File: frontend/src/components/NormReferences.vue [hunk 1]
```diff
@@ -23,23 +24,40 @@ const norms = computed({
   },
 })
 
+function hasSingleNorms(normEntry: NormReference) {
+  if (normEntry.singleNorms)
+    return (
+      normEntry.singleNorms?.length > 1 || !normEntry.singleNorms[0].isEmpty
```
Comment (anne-ds) [inline frontend/src/components/NormReferences.vue:30]: When length is 0 this line will throw an TypeError: Cannot read properties of undefined (reading 'isEmpty').
Use case: Use deletes all singleNorms.

Comment (leonie-koch) [reply frontend/src/components/NormReferences.vue:30]: Gut gesehen! hab ich korrigiert

Comment (rvp-c) [reply frontend/src/components/NormReferences.vue]: But is normEntity.singleNorms not false if it's a empty array?

File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/NormReferenceDTO.java [hunk 2]
```diff
@@ -31,6 +31,9 @@ public class NormReferenceDTO {
   @JoinColumn(name = "norm_abbreviation_id")
   private NormAbbreviationDTO normAbbreviation;
 
+  @Column(name = "norm_abbreviation_raw_value", insertable = false, updatable = false)
+  private String normAbbreviationRawValue;
```
Comment (anne-ds) [inline backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/NormReferenceDTO.java:35]: Is the introduction of the rawValue aligned with the YAGNI-Principle? The frontend is not using it at this point, right?

Comment (rvp-c) [reply backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/database/jpa/NormReferenceDTO.java:35]: But I hope we do it with this refactoring too?

File: frontend/src/components/NormReferenceInput.vue [hunk 3]
```diff
@@ -87,23 +100,15 @@ watch(
     norm.value = new NormReference({ ...props.modelValue })
     lastSavedModelValue.value = new NormReference({ ...props.modelValue })
     if (lastSavedModelValue.value.isEmpty) validationStore.reset()
+    //when list is empty, add new emptry single norm entry
```
Comment (anne-ds) [inline frontend/src/components/NormReferenceInput.vue:103]:
```suggestion
    //when list is empty, add new empty single norm entry
```

File: frontend/src/components/NormReferenceInput.vue [hunk 4]
```diff
@@ -121,97 +126,56 @@ onBeforeUnmount(() => {
...
-            v-if="!lastSavedModelValue.isEmpty"
-            aria-label="Abbrechen"
-            button-type="ghost"
-            label="Abbrechen"
-            size="small"
-            @click.stop="emit('cancelEdit')"
-          />
+    <div v-if="normAbbreviation">
+      <SingleNormInput
+        v-for="(entry, index) in singleNorms"
```
Comment (anne-ds) [inline frontend/src/components/NormReferenceInput.vue:135]:
```suggestion
        v-for="(_, index) in singleNorms"
```

File: frontend/src/components/NormReferences.vue (hunk 1 above)
Comment (anne-ds) [reply frontend/src/components/NormReferences.vue]: Hm, I don't see your fix 🤔 ? Did everything work fine git-wise?

Comment (leonie-koch) [reply frontend/src/components/NormReferences.vue]: This condition seems to work though :)

**Pull request merged**